onnxruntime/core/session/provider_bridge_ort.cc

```diff
@@ -1651,6 +1782,14 @@ static ProviderLibrary s_library_tensorrt(LIBRARY_PREFIX ORT_TSTR("onnxruntime_p
 );
 static ProviderLibrary s_library_migraphx(LIBRARY_PREFIX ORT_TSTR("onnxruntime_providers_migraphx") LIBRARY_EXTENSION);
 
+// QNN EP can be built either as a static library or a shared library. Can safely define s_library_qnn even if static.
+static ProviderLibrary s_library_qnn(LIBRARY_PREFIX ORT_TSTR("onnxruntime_providers_qnn") LIBRARY_EXTENSION
+#ifndef _WIN32
+                                     ,
+                                     false /* unload - On Linux if we unload the qnn shared provider we crash */
```

Comment:
Do we crash or was this just copied from the others?

Reply:
good catch. I meant to test this out. I pushed an experimental commit to see if it crashes on any linux CI pipeline. I'll report back with results.

Reply:
All linux pipelines pass and I checked locally on Ubuntu VM that unload does not crash. Thanks again for catching this.